src/v/cluster/metadata_cache.cc

```diff
@@ -321,8 +321,7 @@ metadata_cache::get_default_record_value_subject_name_strategy() const {
 
 std::optional<std::chrono::milliseconds>
 metadata_cache::get_default_delete_retention_ms() const {
-    // TODO: return config::shard_local_cfg().tombstone_retention_ms();
-    return std::nullopt;
+    return config::shard_local_cfg().tombstone_retention_ms();
```

Comment:
if this PR moves up a few commits in the history then do all the TODO items not need to appear in the PR?

Reply:
Yes, but I felt (at the time) that adding the topic property first then implementing the cluster level config was right.

If you'd prefer a cleaner history by moving the cluster property additions before the topic level property, I can move it. LMKWYT.

Reply:
no need to move it now. i should have caught it earlier in the review